plugins/processors/port_name/port_name.go

```diff
@@ -119,7 +119,15 @@ func (d *PortName) Apply(metrics ...telegraf.Metric) []telegraf.Metric {
 		}
 		if len(d.SourceField) > 0 {
 			if field, ok := m.GetField(d.SourceField); ok {
-				portProto = field.(string)
+				switch v := field.(type) {
+				default:
+					d.Log.Errorf("Unexpected type %t in source field; must be string or int", v)
```

Comment:
I think unsigned ints might be possible as well

Reply:
OK, added.